File: app/src/main/java/com/shuyu/apprecycler/chat/data/factory/ILocalChatDBManager.java
package com.shuyu.apprecycler.chat.data.factory;

import com.shuyu.apprecycler.chat.data.model.ChatBaseModel;

import java.util.List;

import io.reactivex.Observable;

/**
 * 本地数据库接口
 * Created by guoshuyu on 2017/9/11.
 */

public interface ILocalChatDBManager {

    void saveChatMessage(final ChatBaseModel baseModel);

    void getChatMessage(final String chatId, final int page, final ILocalChatDetailGetListener listener);

    void closeDB();
}

File: app-kotlin/src/main/java/example/shuyu/recycler/kotlin/chat/data/factory/ILocalChatDBManager.kt
package example.shuyu.recycler.kotlin.chat.data.factory


import example.shuyu.recycler.kotlin.chat.data.model.ChatBaseModel

/**
 * 本地数据库接口
 * Created by guoshuyu on 2017/9/11.
 */

open interface ILocalChatDBManager {

    fun saveChatMessage(baseModel: ChatBaseModel)

    fun getChatMessage(chatId: String, page: Int, listener: ILocalChatDetailGetListener)

    fun closeDB()
}
